ingesters/namedpipe/namedpipeingester_test.go

```diff
@@ -0,0 +1,73 @@
+package namedpipe_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"syscall"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+
+	"github.com/metal-toolbox/audito-maldito/ingesters/namedpipe"
+	"github.com/metal-toolbox/audito-maldito/internal/health"
+)
+
+func TestIngest(t *testing.T) {
+	t.Parallel()
+
+	tmpDir, err := os.MkdirTemp("", "")
+	if err != nil {
+		t.Errorf("failed to initialize tests: Could not create %s dir", tmpDir)
+	}
+
+	pipePath := fmt.Sprintf("%s/named-pipe", tmpDir)
+	defer func() {
+		os.RemoveAll(tmpDir)
+	}()
+	err = syscall.Mkfifo(pipePath, 0o664)
+	if err != nil {
+		t.Errorf("failed to initialize tests: Could not create %s/%s named pipe", tmpDir, pipePath)
+	}
+
+	sugar := zap.NewExample().Sugar()
+	h := health.NewSingleReadinessHealth("namedpipe")
```

Comment:
the value in the health object initialization needs to match what the `OnReady` call is reporting as ready. With this implementation, the `namedpipe` component will never be marked as ready.